lib/pavilion/plugins/commands/wait.py

```diff
@@ -80,37 +103,47 @@ def run(self, pav_cfg, args):
                     tests.remove(test_id)
 
             # print status every 5 seconds
-            if not args.silent:
-                if time.time() > (start_time + 5*periodic_status_count):
-                    stats = status.get_statuses(pav_cfg, args, self.errfile)
-                    stats_out = []
-
-                    if not args.oneline:
-                        for test in stats:
-                            stat = [str(time.ctime(time.time())), ':',
-                                    'test #',
-                                    str(test['test_id']),
-                                    test['name'],
-                                    test['state'],
-                                    test['note'],
-                                    "\n"]
-                            stats_out.append(' '.join(stat))
-                        fprint(''.join(map(str, stats_out)),
-                               file=self.outfile, width=None)
-                    else:
-                        stats_out.extend([
-                            str(time.ctime(time.time())), ': '
-                        ])
-                        for test in stats:
-                            stat = [str(test['test_id']),
-                                    '(', test['name'], ') ',
-                                    test['state'], ' | ']
-                            stats_out.append(''.join(stat))
-                        fprint(''.join(map(str, stats_out)), end='\r',
-                               file=self.outfile, width=None)
-
-                    periodic_status_count += 1
-
-        final_stats = status.get_statuses(pav_cfg, args, self.errfile)
+            if time.time() > status_time:
+                status_time = time.time() + self.STATUS_UPDATE_PERIOD
+
+                stats = status.get_statuses(pav_cfg, tests, self.errfile)
+                stats_out = []
+
+                if out_mode == self.OUT_SILENT:
+                    pass
+                elif out_mode == self.OUT_ONELINE:
+                    stats_out.extend([str(time.ctime(time.time())), ': '])
+                    for test in stats:
+                        stat = "{t[test_id]}({t[name]}) {t[state]} | "\
+                            .format(t=test)
+                        stats_out.append(stat)
```

Comment:
- Use the time shortening function in output/utils that Jordan wrote. 
- Instead of giving a single message per test, organize by the current test status like the one-line status does when building.  `RUNNING: 5 | COMPLETE: 5 | ` etc...
- Truncate that at the width of the terminal. Terminal width is available through `shutil`.

Reply:
Apparently that's what '--summary' already does.  Nevermind.